homeassistant/components/aemet/weather_update_coordinator.py

```diff
@@ -434,17 +434,13 @@ def _calc_precipitation(self, day, hour):
         rain_value = self._get_rain(day, hour) or 0
         snow_value = self._get_snow(day, hour) or 0
 
-        if round(rain_value + snow_value, 1) == 0:
-            return None
```

Comment:
This will change the precipitation from None to 0 for this case. Is that correct? Ie is a missing value the same as 0 precipitation?

Reply:
The problem is that in the last two lines of code it will directly set it to 0 if the incoming value is None, so given that the intention was to set it to 0 then setting it back to None is not helpful

Also In my opinion it is more "playable" if the value is 0 other than None since None I believe shows "Unknown" in the frontend

Reply:
We represent unknown value with `None` in Home Assistant. The existing code seems correct to me although it may perhaps be longer than needed as you point out. Someone familiar with the Aemet API needs to clarify what missing value means with this API though.

Reply:
> We represent unknown value with `None` in Home Assistant. The existing code seems correct to me although it may perhaps be longer than needed as you point out. Someone familiar with the Aemet API needs to clarify what missing value means with this API though.

I will check this and report back, because I don't remember what the actual behaviour of the API is...

Reply:
If precipitation values are always present we should simplify the code more and remove all the optional return values for those code paths.